cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/index.tsx

```diff
@@ -0,0 +1,178 @@
+
+const styles = (theme): StyleRules => {
+  return {
+    eachWidget: {
+      display: 'grid',
+      gridTemplateColumns: '5fr 1fr',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    },
+    widgetInputs: {
+      '& > *': {
+        marginTop: '20px',
+        marginBottom: '20px',
+      },
+    },
+    nestedWidgets: {
+      border: `1px solid`,
+      borderColor: theme.palette.grey[300],
+      borderRadius: '6px',
+      position: 'relative',
+      padding: '7px 10px 5px',
+      margin: '25px',
+    },
+    addWidgetLabel: {
+      fontSize: '12px',
+      position: 'absolute',
+      top: '-10px',
+      left: '15px',
+      padding: '0 5px',
+      backgroundColor: theme.palette.white[50],
+    },
+    required: {
+      fontSize: '14px',
+      marginLeft: '5px',
+      lineHeight: '12px',
+      verticalAlign: 'middle',
+    },
+    widgetContainer: {
+      width: 'calc(100%-1000px)',
+    },
+    widgetDivider: {
+      width: '100%',
+    },
+  };
+};
+
+interface IWidgetCollectionProps extends WithStyles<typeof styles>, ICreateContext {
+  groupID: number;
+}
+
+const WidgetCollectionView: React.FC<IWidgetCollectionProps> = ({
+  classes,
+  groupID,
+  widgetToInfo,
+  groupToWidgets,
+  setWidgetToInfo,
+  setGroupToWidgets,
+}) => {
+  const [activeWidgets, setActiveWidgets] = React.useState(groupID ? groupToWidgets[groupID] : []);
+
+  React.useEffect(() => {
+    if (groupID) {
+      setActiveWidgets(groupToWidgets[groupID]);
+    } else {
+      setActiveWidgets([]);
+    }
+  }, [groupToWidgets]);
+
+  function addWidgetToGroup(index: number) {
+    return () => {
+      const newWidgetID = 'Widget_' + uuidV4();
+
+      // Add a new widget's ID at the specified index
+      const newWidgets = [...activeWidgets];
+      if (newWidgets.length === 0) {
+        newWidgets.splice(0, 0, newWidgetID);
+      } else {
+        newWidgets.splice(index + 1, 0, newWidgetID);
+      }
+      setGroupToWidgets({
+        ...groupToWidgets,
+        [groupID]: newWidgets,
+      });
+
+      // Set the default empty properties of the widget
+      setWidgetToInfo({
+        ...widgetToInfo,
+        [newWidgetID]: {
+          widgetType: '',
+          label: '',
+          name: '',
+        } as IWidgetInfo,
+      });
+    };
+  }
+
+  function deleteWidgetFromGroup(widgetIndex) {
+    return () => {
+      // Grab the widget ID to delete
+      const widgetToDelete = activeWidgets[widgetIndex];
+
+      const newWidgets = [...activeWidgets];
+      newWidgets.splice(widgetIndex, 1);
+      setGroupToWidgets({
+        ...groupToWidgets,
+        [groupID]: newWidgets,
+      });
+
+      const { [widgetToDelete]: info, ...restWidgetToInfo } = widgetToInfo;
+      setWidgetToInfo(restWidgetToInfo);
+    };
+  }
+
+  return (
+    <div className={classes.nestedWidgets} data-cy="widget-wrapper-container">
+      <div className={`widget-wrapper-label ${classes.addWidgetLabel}`}>
```

Comment:
Can we pull this style into this component and remove plain css or scss usage?

Reply:
Changed!